python/tests/test_aac/context/test_definition.py

```diff
@@ -18,10 +19,56 @@ def test_get_root_key(self):
 
         definition = context.get_definitions_by_name("abstract")
         self.assertEqual(definition[0].get_root_key(), "modifier")
-        
+
     def test_to_yaml(self):
         context = LanguageContext()
         definition = context.get_definitions_by_name("Schema")
         yaml = definition[0].to_yaml()
         self.assertIn("schema:", yaml)
         self.assertIn("name: Schema", yaml)
+
+    def test_definition_fail_incorrect_types(self):
+        context = LanguageContext()
+        schema_definition = context.get_definitions_by_name("Schema")[0]
+        with self.assertRaises(TypeError):
+            # This is probably the worst way to make a definition, it should normally be handled by Language Context
+            definition = Definition(
+                name=123,
+                package=schema_definition.package,
+                content=schema_definition.content,
+                source=schema_definition.source,
+                structure=schema_definition.structure,
+            )
+        with self.assertRaises(TypeError):
+            definition = Definition(            # noqa: F841
+                name=schema_definition.name,
+                package=123,
+                content=schema_definition.content,
+                source=schema_definition.source,
+                structure=schema_definition.structure,
+            )
```

Comment:
Agree with your comment, but good to check that the class constructor does properly handle, even if it is "managed" by a different layer.

Can test the content piece with bad yaml content possibly?